pylearn2/datasets/dataset.py

```diff
@@ -18,7 +18,7 @@ def __iter__(self):
         return self.iterator()
 
```

Comment:
That is what I meant, they should disappear from the interface of `Dataset`, but can still be part of `DenseDesignMatrix`.